configs/sample_yamls/serviceops/sample_service_config.yml

```diff
@@ -0,0 +1,172 @@
+#
+########################## types ########################
+#
+types:
```

Comment:
It seems we're all in agreement that type inference is possible from constructors without a loss of fidelity.

Just to be clear you're advocating delaying doing that work until you've gotten other things working? That's reasonable although I do have concerns about the user-experience of authoring config which is what I was trying to focus on in my comments. 

Its pretty clear that there's some collateral damage to the samples in this PR because of naming confusion (see my other comments). This seems to be caused by the semantic overlap of constructor and type. I wouldn't want to live with this in the  product if it can be avoided.